src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,52 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.UserRegisterException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+
+    public static final int MIN_LENGTH_PASSWORD_PARAMETER = 6;
+    public static final int MIN_LENGTH_LOGIN_PARAMETER = 6;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUserExist(user);
+        checkUserLoginLength(user);
+        checkUserPassLength(user);
+        checkUserAge(user);
+        return storageDao.add(user);
```

Comment:
One of the checklist points is 'Don't create one big if with all conditions. Better to split one big if into several small ones.' But in your case, you have created separate methods for each condition. This is also a good approach because it makes your code more readable and maintainable. However, you should also consider the performance implications of calling multiple methods. In general, it's a trade-off between readability and performance.

Reply:
Very interesting...